pkg/state/state_nv_ipam_cni_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+2024 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
```

Comment:
We should implement test in a different package and call only public APIs.

See for example:
https://github.com/Mellanox/network-operator/blob/master/pkg/state/state_macvlan_network_test.go

Reply:
+1 thats where we want to be at.

Reply:
Thanks @adrianchiris for clearing up the _test package issue - I'll update to move the tests there.

Reply:
Moved the tests to the `state_test` pkg